chainerx_cc/chainerx/routines/math.cc

```diff
@@ -1110,4 +1110,30 @@ Array IsInf(const Array& x) {
     return out;
 }
 
+template <typename Impl>
+inline Array BitwiseImpl(Impl&& impl, const Array& x1, const Array& x2) {
+    CHAINERX_ASSERT(GetKind(x1.dtype()) != DtypeKind::kFloat);
+    CHAINERX_ASSERT(GetKind(x2.dtype()) != DtypeKind::kFloat);
+    // promote to which ever is bigger
+    Array x1_cast = x1.dtype() < x2.dtype() ? x1.AsType(x2.dtype()) : x1;
+    Array x2_cast = x2.dtype() < x1.dtype() ? x2.AsType(x1.dtype()) : x2;
+    Dtype out_dtype = x1_cast.dtype();
```

Comment:
We can write it simpler as follows:
```
Dtype out_dtype = std::max(x1_dtype(), x2_dtype());
const Array& x1_cast = ...
const Array& x2_cast = ...
```
However, I think it is confusing that the dtype conversion rule for bitwise operations is different from that for other arithmetic operations.

Reply:
I discussed it with a team member and we decide to use the same dtype rule as other arithmetic operations. Could you use `GetArithmeticResultDtype`?

Reply:
Oh sure.